framework/platform/bpl/linux/bpl_cfg.cpp

```diff
@@ -391,5 +391,45 @@ bool cfg_get_unfriendly_device_max_timelife_delay_days(
     return true;
 }
 
+bool bpl_cfg_get_wpa_supplicant_ctrl_path(const std::string &iface, std::string &wpa_ctrl_path)
+{
+
+    std::string local_wpa_ctrl_path;
+    std::string param = "wpa_supplicant_ctrl_path_";
+
+    param += iface;
+    param += '=';
+
+    if (cfg_get_param(param, local_wpa_ctrl_path) < 0) {
+        MAPF_ERR("bpl_cfg_get_wpa_supplicant_ctrl_path: Failed to read: " << param);
+        return false;
+    }
+
+    std::copy(local_wpa_ctrl_path.begin(), local_wpa_ctrl_path.end(),
+              std::back_inserter(wpa_ctrl_path));
```

Comment:
Why not follow next way:
```
wpa_ctrl_path = local_wpa_ctrl_path;
```
It will also create string copy:
https://en.cppreference.com/w/cpp/string/basic_string/operator%3D
```
basic_string& operator=( const basic_string& str );   (1)
....
1) Replaces the contents with a copy of str. If *this and str are the same object, this function has no effect.
```

Reply:
fixed